ls1intum/Artemis
# `Assessment`: Fix an issue with the complaint timeframe ending too early

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


Currently there is a bug that students cannot complain the whole time frame that they are entitled to. This PR fixes this.

### Description

Before this PR results with a completion date before the due date of an exercise used this as the starting time, while it should be the due date or assessment duedate, depending on which exists and which is later in time.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Exercise in a course where complaints are possible

1. Log in to Artemis
2. Find an exercise with an old result without a complaint
3. Change the due date and/or assessment due date to 1-2 days (make sure that the course allows complaints for at least three days) before now
4. Make sure that you can complain in this exercise now

#### Exam Mode Testing


Only changed code where it was checked that it's a course exercise

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| complaints-student-view.component.ts | 77% | 94% |
